homeassistant/components/logbook.py

```diff
@@ -97,6 +118,14 @@ def humanify(events):
             if event.event_type == EVENT_STATE_CHANGED:
                 entity_id = event.data['entity_id']
 
+                # This should be rafactored so that the SQL query does
+                # the filtering, possibly by adding an option entity_id
+                # field to events
```

Comment:
Note, this has been fixed now in the `dev` branch. An event_id column was added to the states table that points at the related row in the events table.